src/messages/MClientRequest.h

```diff
@@ -75,6 +75,106 @@ struct SnapPayload {
 
 WRITE_CLASS_ENCODER(SnapPayload)
 
+struct NotificationEndpointPayload {
+  virtual void encode(ceph::buffer::list &bl) const = 0;
+  virtual void decode(ceph::buffer::list::const_iterator &iter) = 0;
+  virtual void dump(ceph::Formatter *f) const = 0;
+  virtual ~NotificationEndpointPayload() {}
+};
+
+struct KafkaTopicPayload final : public NotificationEndpointPayload {
+  std::string topic_name;
+  std::string broker;
+  bool use_ssl;
```

Comment:
This could remain unassigned, please set default value either here  or in all the ctors.